Iter5

SinnerUfa opened the pull request.

SinnerUfa committed:
Итерация 1 -  сервер + Итерации 4, 5 (флаги и окружение) + вывод страницы со всеми метриками

SinnerUfa committed:
Начало работы с автотестом

SinnerUfa committed:
Исправление ошибок найденных go vet

SinnerUfa committed:
Исправление вновь найденных ошибок go vet

SinnerUfa committed:
Исправление ошибок найденных go vet

SinnerUfa committed:
Проверка влияния sync

SinnerUfa committed:
Итерация 2, сделать проверку автотестами

SinnerUfa committed:
Итер3 добавлено закомментирование формаирование страницы на адресе сервера (необходимо посмотреть тесты)

SinnerUfa committed:
Итер 3 странность с обработкой корневого адреса (проверка завершилась на итер 1) необходим возврат и просмотр тестов

SinnerUfa committed:
Итер 3 обработка float

SinnerUfa committed:
Итер3 проблема float

SinnerUfa committed:
Итер3 проблема float

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема сторонних пакетов

SinnerUfa committed:
Итер3 проблема ошибок

SinnerUfa committed:
Итер3 проблема ошибок

SinnerUfa committed:
Итер3 проблема ошибок

SinnerUfa committed:
Итер3 проблема главной страницы

SinnerUfa committed:
Итер4 пробуем тесты

SinnerUfa committed:
Итер5 очистка комментариев, пробуем тесты

SinnerUfa committed:
Итер5 исправление имени окружения, пробуем тесты

tmvrus commented on cmd/agent/main.go:
```diff
@@ -1,3 +1,38 @@
+	log := mlog.New(true)
+
+	if err := config.Load(&cfg, os.Args[1:]); err != nil {
+		log.Error(err)
```
разве нормально продолжить работу если случилась подобная ошибка?

SinnerUfa replied:
[+]

tmvrus commented on cmd/agent/main.go:
```diff
@@ -1,3 +1,38 @@
+	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt, os.Kill)
+
+	if err := run(ctx, log, cfg); err != nil {
+		log.Info(err)
```
log.Error?

SinnerUfa replied:
[+]

tmvrus commented on cmd/server/main.go:
```diff
@@ -1,3 +1,38 @@
+	log := mlog.New(true)
+
+	if err := config.Load(&cfg, os.Args[1:]); err != nil {
+		log.Error(err)
```
и тут

SinnerUfa replied:
[+]

tmvrus commented on internal/agent/agent.go:
```diff
@@ -0,0 +1,28 @@
+	poster := NewPoster(ctx, log, rep, cfg.Adress)
+	ticker.NewAndRun(ctx, cfg.ReportInterval, poster)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(log mlog.Logger) {
+		defer wg.Done()
+		<-ctx.Done()
+		log.Info("stop periodic functions")
+	}(log)
+	wg.Wait()
```
- а зачем использовать wg если горутина только одна?
- если код все время ждет завершения одной горутины, может она не нужна?
- "stop periodic functions" как бы говорит что у ф-и должны быть какие периоды

SinnerUfa replied:
[+]

tmvrus commented on internal/agent/poster.go:
```diff
@@ -0,0 +1,56 @@
+
+	for _, v := range l {
+		endpoint := "http://" + m.adress + "/" + v.ReguestString("update")
+		// request, err := http.NewRequestWithContext(m.ctx, http.MethodPost, endpoint, http.NoBody)
+		// if err != nil {
+		// 	return err
+		// }
+		// request.Header.Add("Content-Type", "text/plain")
+
+		// response, err := client.Do(request)
+		// if err != nil {
+		// 	return nil // нужно подумать
+		// }
+		// defer response.Body.Close()
+		// _, err = io.ReadAll(response.Body)
+		// if err != nil {
+		// 	return nil
+		// }
```
вот это дурной тон, засоряет код и усложняет его чтение

SinnerUfa replied:
[+]

tmvrus commented on internal/agent/poster.go:
```diff
@@ -0,0 +1,56 @@
+		// 	return nil
+		// }
+		_, err := client.R().SetHeader("Content-Type", "text/plain").Post(endpoint)
+		if err != nil {
+			return nil
+		}
```
?

SinnerUfa replied:
[+]

tmvrus commented on internal/server/hundlers/get_list.go:
```diff
@@ -0,0 +1,27 @@
+	return http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			t, err := template.ParseFiles("index.html", "body_list.html")
+			if err != nil {
+				http.Error(w, codes.ErrRepNotFound.Error(), http.StatusInternalServerError)
+				log.Warning(codes.ErrGetLstParse)
```
тут видимо return потерялся

SinnerUfa replied:
[+]

tmvrus commented on internal/server/server.go:
```diff
@@ -0,0 +1,36 @@
+		}
+	}(log)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(log mlog.Logger) {
+		defer wg.Done()
+		<-ctx.Done()
+		if err := httpServer.Shutdown(ctx); err != nil {
+			log.Error("error shutdown server: ", err)
+		}
+	}(log)
+	wg.Wait()
```
вот тут тоже неясно зачем горутина и зачем wg

SinnerUfa replied:
[+]

tmvrus reviewed (changes requested):
есть вопросы

SinnerUfa committed:
Спринт 1  - Исправление замечаний ревью 1
Замечания:
1. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635381233]
2. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635381396]

Исправления:
1. Удален блок кмментариев
2. Сделано логирование и возврат ошибки

SinnerUfa committed:
Спринт 1  - Исправление замечаний ревью 1
Замечания:
1. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635384982]

Исправления:
1. Вставлен забытый возврат из функции

SinnerUfa committed:
Спринт 1  - Исправление замечаний ревью 1
Замечания:
1. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635385943]
2. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635377452]

Исправления:
1. Удален блок WaitGroup (данный блок от собственного шаблона сервиса)
2. Удален блок WaitGroup (данный блок от собственного шаблона сервиса)

SinnerUfa committed:
Спринт 1  - Исправление замечаний ревью 1
Замечания:
1. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635374797]
2. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635375062]
3. [https://github.com/SinnerUfa/practicum-metrics/pull/5#discussion_r1635375496]

Исправления:
1, 2, 3.
    - Методы логгера Crit и Error приведены к названиям методо стандартного логера (Fatal и Panic соотвественно) и исправлены все вызовы;
    - Вызовы метода логгера Eror изменены на Warning + возвраты из функций;
    - Ошибка сервера помещена в канал и передается на более высокий уровень;
    - Ошибки сервера внесены в общий список ошибок;

tmvrus commented on cmd/agent/main.go:
```diff
@@ -1,3 +1,39 @@
+	log := mlog.New(true)
+
+	if err := config.Load(&cfg, os.Args[1:]); err != nil {
+		log.Warning(err)
+		os.Exit(1)
```
но есть же `Fatal(v ...any)`

tmvrus commented on cmd/agent/main.go:
```diff
@@ -1,3 +1,39 @@
+	if err := run(ctx, log, cfg); err != nil {
+		log.Warning(err)
+		cancel()
+		os.Exit(1)
```
тоже самое

tmvrus commented on internal/config/config.go:
```diff
@@ -0,0 +1,24 @@
+	if err := env.Load(v); err != nil {
+		return err
+	}
+	// if err:=validate(v);err!=nil{
+	//     return err
+	// }
```
не стоит оставлять закоментированный код, он зашумляет проект и усложняет его чтение
